app/src/main/java/ch/epfl/sdp/appart/ad/UserAdsViewModel.java

```diff
@@ -0,0 +1,62 @@
+package ch.epfl.sdp.appart.ad;
+
+import android.util.Log;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import javax.inject.Inject;
+
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+import ch.epfl.sdp.appart.database.DatabaseService;
+import ch.epfl.sdp.appart.login.LoginService;
+import ch.epfl.sdp.appart.scrolling.card.Card;
+import ch.epfl.sdp.appart.user.User;
+import dagger.hilt.android.lifecycle.HiltViewModel;
+
+@HiltViewModel
+public class UserAdsViewModel extends ViewModel {
+    private final MutableLiveData<List<Card>> lUserAds = new MutableLiveData<>();
+    final DatabaseService database;
+    final LoginService loginService;
+
+    @Inject
+    public UserAdsViewModel(DatabaseService database, LoginService loginService) {
+        this.database = database;
+        this.loginService = loginService;
+    }
+
+    public void initHome() {
+        CompletableFuture<User> user = database.getUser(loginService.getCurrentUser().getUserId());
+        user.exceptionally(e -> {
+            Log.d("EXCEPTION_DB", e.getMessage());

```

Comment:
Both the exceptionally in this class aren't tested, with the help of the mock database you could test these lines and reach the 80% coverage target

Reply:
I don't really see how I could test that, as it would require the mockdatabase to fail to query the current user/ cards. Do you have an idea how to do so ?

Reply:
What I did on other cases is to fail given a specific id. For example, you can log in with a specific user (say with id "0000"), and when the mockdatabase is queried for a user with id "0000" it completes exceptionally.
It is not super clean, but it's the only easy way I see to these the exceptionally blocks. If there were just few of this blocks it wouldn't be so important to test them, but in our project there are so many of them that we risk to lose  a lot of coverage if we do not test them.